core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala

```diff
@@ -97,7 +99,15 @@ private[v1] class AbstractApplicationResource extends BaseAppResource {
 
   @GET
   @Path("environment")
-  def environmentInfo(): ApplicationEnvironmentInfo = withUI(_.store.environmentInfo())
+  def environmentInfo(): ApplicationEnvironmentInfo = withUI { ui =>
+    val envInfo = ui.store.environmentInfo()
+    new v1.ApplicationEnvironmentInfo(
+      envInfo.runtime,
+      Utils.redact(ui.conf, envInfo.sparkProperties),
+      Utils.redact(ui.conf, envInfo.hadoopProperties),
+      Utils.redact(ui.conf, envInfo.systemProperties),
+      Utils.redact(ui.conf, envInfo.classpathEntries))
```

Comment:
As in the other PR, I don't know if we need to redact the classpath?

Reply:
+1. I'm not sure what's the point of applying redact to classpath entries, and why exposing classpath entries without redacting is considered as security issue. Moreover I'm not sure what would be matched to redact patterns from classpath entries.

Reply:
jar name containing key words like password, token etc.. i have updated the PR to skip redacting classpath.